app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php

```diff
@@ -146,7 +169,8 @@ protected function validateEmailFormat($email)
      */
     public function execute()
     {
-        if ($this->getRequest()->isPost() && $this->getRequest()->getPost('email')) {
+        if ($this->validateIsSubscriptionEnabled() &&
+            $this->getRequest()->isPost() && $this->getRequest()->getPost('email')) {
```

Comment:
Please move `validateIsSubscriptionEnabled` at the end and correctly format the multiline if condition. Check [this article](https://pear.php.net/manual/en/rfc.cs-enhancements.splitlongstatements.php) for more info.
```suggestion
        if ($this->validateIsSubscriptionEnabled() &&
            $this->getRequest()->isPost() && $this->getRequest()->getPost('email')) {
```